ui/src/diary/diary-add-note.tsx

```diff
@@ -16,18 +16,24 @@ import {
 } from '@/state/diary';
 import { useChannel, useGroup, useRouteGroup } from '@/state/groups';
 import { DiaryBlock, NoteContent, NoteEssay } from '@/types/diary';
-import { Inline } from '@/types/content';
+import { Inline, JSONContent } from '@/types/content';
 import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';
 import PencilIcon from '@/components/icons/PencilIcon';
 import { useIsMobile } from '@/logic/useMedia';
 import ReconnectingSpinner from '@/components/ReconnectingSpinner';
 import useGroupPrivacy from '@/logic/useGroupPrivacy';
 import { captureGroupsAnalyticsEvent } from '@/logic/analytics';
+import Toggle from '@/components/Toggle';
 import DiaryInlineEditor, { useDiaryInlineEditor } from './DiaryInlineEditor';
+import DiaryMarkdownEditor from './DiaryMarkdownEditor';
 
 export default function DiaryAddNote() {
   const { chShip, chName, id } = useParams();
   const [loaded, setLoaded] = useState(false);
+  const [editWithMarkdown, setEditWithMarkdown] = useState(true);
+  const [updateMarkdown, setUpdateMarkdown] = useState(false);
+  const [updateTipTap, setUpdateTipTap] = useState(true);
+  const [editorContent, setEditorContent] = useState<JSONContent | null>(null);
```

Comment:
the logic around updating with these flags seems like it would be easy to mess up. can we streamline this further or make these unnecessary somehow? like maybe we always have both the editor content and the markdown contents ready so flipping the switch is just really visual?